flink-runtime/src/main/java/org/apache/flink/runtime/externalresource/ExternalResourceUtils.java

```diff
@@ -157,6 +167,30 @@ private ExternalResourceUtils() {
         return externalResourceAmountMap;
     }
 
+    /** Get the collection of all enabled external resources. */
+    public static Collection<ExternalResource> getExternalResourcesCollection(
+            Configuration config) {
+        return getExternalResourceAmountMap(config).entrySet().stream()
+                .map(entry -> new ExternalResource(entry.getKey(), entry.getValue()))
+                .collect(Collectors.toList());
+    }
+
+    /** Generate the string expression of the given external resources. */
+    public static String generateExternalResourcesString(
+            Map<String, ExternalResource> extendedResources) {
+        final StringBuilder extendedResourceStr = new StringBuilder(extendedResources.size() * 10);
+        for (Map.Entry<String, ExternalResource> resource : extendedResources.entrySet()) {
+            if (extendedResourceStr.length() > 0) {
+                extendedResourceStr.append(", ");
+            }
+            extendedResourceStr
+                    .append(resource.getKey())
+                    .append('=')
+                    .append(resource.getValue().getValue());
+        }
```

Comment:
This can be simplified as follows:
```
return extendedResources.entrySet().stream()
        .map(e -> e.getKey() + "=" + e.getValue().getValue())
        .collect(Collectors.joining(", "));
```